### PR TITLE
Add common event format plugin

### PR DESCRIPTION
**Pull request by ericwholt**
This plugin adds option to parse common event format logs. Currently it is not possible to parse the extensions without the addition of an operator. Supports syslog prefix with timestamp matching `Jan 02 15:04:05 ` and a hostname.

### CODE REVIEW

**Review by jsirianni — CHANGES_REQUESTED** [review]
Some thoughts
```
name --> message
device_ --> labels
hostname --> resource
signature. --> ??
version --> ??
```

Also, I will be building a key value parser, so we can handle extenssions.

File: plugins/common_event_format.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  # Timestamp Jan 02 15:04:05 - Parse syslog prefix. 
+  # If more than one timestamp parser is required then it will need to be move to a stand alone timestamp_parser
+  - id: timestamp_parser
+    type: regex_parser
+    parse_from: $record
+    regex: '^(?P<timestamp>\w{3}\s+\d{1,2}\s+\d{2}:\d{2}:\d{2})\s+((?P<hostname>[^\s]+)\s+)?(?P<cef_headers>[\d\D]+)'
+    timestamp:
+      parse_from: timestamp
+      layout_type: gotime
+      layout: 'Jan 02 15:04:05'
```
[the remark this answers is not in the document]
Comment (ericwholt) [reply plugins/common_event_format.yaml:90]: Done

**Comment by ericwholt** [comment]
Removed key_value parser. Extensions format allows to many variations to parse using the parser.

**Comment by jsirianni** [comment]
Lets promote device_vendor / device_version to resources

**Review by jsirianni — APPROVED** [review]

**Comment by jsirianni** [comment]
```json
{
  "timestamp": "2021-09-11T08:26:10Z",
  "severity": 70,
  "severity_text": "10",
  "labels": {
    "device": "threatmanager",
    "file_name": "cef.log",
    "log_type": "cef",
    "plugin_id": "common_event_format",
    "version": "0"
  },
  "resource": {
    "device_vendor": "security",
    "device_version": "1.0",
    "hostname": "zurich"
  },
  "record": {
    "extensions": "src=10.0.0.1 dst=2.1.2.2 spt=1232 ",
    "message": "worm successfully stopped",
    "signature_id": "100"
  }
}
```

**Pull request merged**